### PR TITLE
Rdonat/helm retry

### PR DESCRIPTION
**Pull request by bobdonat**
# Description

The pull request changes the install scripts to detect slow image pulls after a helm install fails.  If slow image pulls are detected then the helm install will be retried after cleaning up the failed helm install.

Fixes VZ-3263

# Checklist 

As the author of this PR, I have:

- [x] Checked that I included or updated copyright and license notices in all files that I altered
- [ ] Added or updated unit tests for any new functions I added
- [ ] Added or updated integration tests if appropriate
- [ ] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

File: platform-operator/scripts/install/common.sh [hunk 1]
```diff
@@ -373,6 +375,52 @@ function generate_password() {
...
+  local max_retries=1
+  while true ; do
+    log "Installing ${ns}/${chart_name}"
+    helm upgrade ${chart_name} ${chart_dir} \
+      --install --namespace ${ns} \
+      --wait --timeout 10m \
+      "$@" && break
+    local helm_status=$?
+    if [ "$retries" -eq "$max_retries" ] ; then
+fix      return $helm_status
```
Comment (bhatpmk) [inline platform-operator/scripts/install/common.sh:415]: @bobdonat Is "fix" something your forgot to remove ?

Comment (bobdonat) [reply platform-operator/scripts/install/common.sh:415]: not sure what that is doing there.  I will remove that.

Comment (bobdonat) [reply platform-operator/scripts/install/common.sh]: pushed change

Comment (bobdonat) [reply platform-operator/scripts/install/common.sh]: thanks for catching that

File: platform-operator/scripts/install/2-install-system-components.sh [hunk 2]
```diff
@@ -406,14 +396,26 @@ function install_rancher()
...
 
-      # Do not add --wait since helm install will not fully work in OLCNE until MKNOD is added in the next command
-      helm upgrade ${chart_name} ${RANCHER_CHART_DIR} \
-        --install --namespace cattle-system \
-        --set hostname=rancher.${NAME}.${DNS_SUFFIX} \
-        --set ingress.tls.source=${INGRESS_TLS_SOURCE} \
-        ${HELM_IMAGE_ARGS} \
-        ${IMAGE_PULL_SECRETS_ARGUMENT} \
-        ${EXTRA_RANCHER_ARGUMENTS}
+      if [ $(is_external_dns) == "true" ]; then
```
Comment (fredtibbitts) [inline platform-operator/scripts/install/2-install-system-components.sh:399]: Does external DNS always mean OLCNE? When I first looked at this I was confused but then it made more sense when I looked at the `is_external_dns` function. Maybe a comment would help?

File: platform-operator/scripts/install/1-install-istio.sh [hunk 3]
```diff
@@ -84,24 +84,20 @@ function install_istio()
     fi
 
     if ! is_chart_deployed istio-base istio-system ${ISTIO_CHART_DIR}/base ; then
-      log "Installing Istio base"
+      log "Installing istio-system/istio-base"
       helm upgrade istio-base ${ISTIO_CHART_DIR}/base \
```
Comment (fredtibbitts) [inline platform-operator/scripts/install/1-install-istio.sh:88]: A comment explaining why we don't need to use `helm_install_retry` here might be good, but I wouldn't add it unless you have other changes to push.

File: platform-operator/scripts/install/2-install-system-components.sh (hunk 2 above)
Comment (bobdonat) [reply platform-operator/scripts/install/2-install-system-components.sh:399]: I think external DNS means OLCNE today.

File: platform-operator/scripts/install/1-install-istio.sh (hunk 3 above)
Comment (bobdonat) [reply platform-operator/scripts/install/1-install-istio.sh:88]: Do you think we should use helm_install_retry here for consistency?

Comment (bobdonat) [reply platform-operator/scripts/install/1-install-istio.sh:88]: even though there are no images in the istio-base helm chart

Comment (fredtibbitts) [reply platform-operator/scripts/install/1-install-istio.sh:88]: Actually yes, it's probably best to just make it consistent.

**Review by fredtibbitts — APPROVED** [review]

**Pull request merged**